roles/confluent.variables/vars/main.yml

```diff
@@ -1115,3 +1115,123 @@ control_center_final_properties: "{{ control_center_combined_properties | combin
+  interceptors_config_file: "{{ archive_config_base_path if installation_method == 'archive' else '' }}/etc/kafka-connect-replicator/kafka-connect-replicator-interceptors.properties"
+  systemd_file: /lib/systemd/system/{{kafka_connect_replicator_service_name}}.service
+  systemd_override: /etc/systemd/system/{{kafka_connect_replicator_service_name}}.service.d/override.conf
+  log4j_file: "{% if installation_method == 'archive' %}{{archive_destination_path}}/confluent-{{confluent_package_version}}{% endif %}/etc/kafka-connect-replicator/replicator-log4j.properties"
+  cluster_id: "{{kafka_connect_replicator_cluster_id}}"
+
+kafka_connect_replicator_properties:
+  defaults:
+    enabled: true
+    properties:
+      listeners: "{{kafka_connect_replicator_http_protocol}}://0.0.0.0:{{kafka_connect_replicator_port}}"
+      rest.advertised.listener: "{{kafka_connect_replicator_http_protocol}}"
+      rest.advertised.host.name: "{{inventory_hostname}}"
+      rest.advertised.port: "{{kafka_connect_replicator_port}}"
+      confluent.topic.bootstrap.servers: "{{kafka_connect_replicator_bootstrap_servers}}"
+      offset.start: "{{kafka_connect_replicator_offset_start}}"
+      offset.storage.topic: "{{kafka_connect_replicator_offsets_topic}}"
+      status.storage.topic: "{{kafka_connect_replicator_status_topic}}"
+      config.storage.topic: "{{kafka_connect_replicator_storage_topic}}"
+      topic.whitelist: "{{kafka_connect_replicator_white_list}}"
+      topic.auto.create: "{{kafka_connect_replicator_topic_auto_create}}"
+      confluent.license.topic: _confluent-command
+  ssl:
+    enabled: kafka_connect_replicator_listener ['ssl_enabled'] | default(ssl_enabled) | bool
+    properties:
+      listeners.https.ssl.keystore.location: "{{kafka_connect_replicator_keystore_path}}"
+      listeners.https.ssl.keystore.password: "{{kafka_connect_replicator_keystore_storepass}}"
+      listeners.https.ssl.key.password: "{{kafka_connect_replicator_keystore_keypass}}"
+      listeners.https.ssl.truststore.location: "{{kafka_connect_replicator_truststore_path}}"
+      listeners.https.ssl.truststore.password: "{{kafka_connect_replicator_truststore_storepass}}"
+  kafka_client:
+    enabled: true
+    properties: "{{ kafka_connect_replicator_listener | client_properties(ssl_enabled, pkcs12_enabled, ssl_mutual_auth_enabled, sasl_protocol,
+                '', kafka_connect_replicator_truststore_path, kafka_connect_replicator_truststore_storepass, kafka_connect_replicator_keystore_path,
+                kafka_connect_replicator_keystore_storepass, kafka_connect_replicator_keystore_keypass, false, kafka_connect_replicator_sasl_plain_principal, kafka_connect_replicator_sasl_plain_password,
+                kafka_connect_replicator_sasl_scram_principal, kafka_connect_replicator_sasl_scram_password, kerberos_kafka_broker_primary, kafka_connect_replicator_keytab_path,
```

Comment:
are we going to allow for unique sasl plain and scram users/passwords for each kafka since you allow for unique certs and keytabs?

Reply:
actually looks like you did that by making variables. maybe this should be added in a test case

Reply:
I originally had this in the test case, with the previous non client code, and then removed it to clean it up as you wanted the client code used, and you wanted to not have to duplicate vars in the molecule.yml.

Reply:
I must have missed what you were going for in the test. In your code there is the feature to connect to each kafka using unique authentication for each (ie one sasl user/pass for one kafka and a diff sasl user/pass for another), but i don't think the tests validate that yet. It looks like they are using the same certs or the same kerberos keytab.

Since unique auth per kafka is something you are offering as a feature, a test should be added that validates this.

Reply:
I disagree with the assessment that this is not being tested.  All of the mechanisms are being fully used, for example we are using a scram username and scram password, this validates the logic works.  Changing the contents of what is inside the field is irrelevant.  It's the same with TLS.

In fact, we actually had a conversation where we discussed how to test the mechanisms without having a sprawling set of test cases where we agreed that the current approach made sense.  I still stand by that it makes sense as different content, doesn't change the mechanism itself.

Reply:
I apologize if I changed my mind on something. This is a big PR that has gone thru many revisions and I am doing my best to review it.

My point is simply you offer the feature for unique auth into each kafka and the tests don't prove that out. I'm not saying the code is incorrect, im just saying that I do not see a test to definitively say that feature works. For example a test that says kafka one has sasl plain, kafka 2 has sasl plain but replicator needs to pass different user/passwords to get into each. 

Its only because you have made auth so customizable that I would like to see it in a test

Reply:
Understood.  If you feel this strongly about additional testing, I would suggest submitting a PR with enhancements after this PR is merged.  Happy to discuss further with management.

Reply:
sure additional test cases do not need to hold up this merge